main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -884,13 +884,23 @@ object Weeder {
 
     case ParsedAst.Expression.ApplicativeFor(sp1, frags, exp, sp2) =>
       val loc = mkSL(sp1, sp2).asSynthetic
-      val fs = traverse(frags)(visitForFragmentGenerator(_))
+      val (gens, bad) = frags.partition {
```

Comment:
Whats happening here with the partition?

Reply:
Filtering out non-generator fragments. This is related to the same question Matt had above, but this is just to enforce the restrictions that were already there (from the Parser).